Repositories/Base/Repository.cs

```diff
@@ -0,0 +1,71 @@
+﻿using EFCoreServices.Models;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Linq;
+using EFCoreServices.DTO;
+
+namespace EFCoreServices.Repository
+{
+    public class Repository : IRepository<Product>
+    {
+        ProductDBContext db;
+        public Repository(ProductDBContext _DbContext)
+        {
+            db = _DbContext;
+        }
+
+        public async Task AddAsync(Product product)
+        {
+            
+                await db.AddAsync(product);
+                await db.SaveChangesAsync();
+        }
+
+        public async Task<int> DeleteAsync(int productID)
+        {
+
+                var product = await db.Products.SingleOrDefaultAsync(e => e.Id == productID);
```

Comment:
throw an exception that you can catch in the controller when the id is not found and return not found